### PR TITLE
Add PCCM radio to MCPAR dashboard create modal

### PR DESCRIPTION
**Pull request by gmrabian**
### Description

At this time we have only added the radio and its metadata. It does not affect the actual report at all. That will be covered separately.

Feedback needed:
We probably want to make this disabled (read-only) after creation, right? So it should only be editable when first creating a report but not if you go back to change the title or dates.

Open to different naming ideas for any variables

### Related ticket(s)

MDCT-3027

---
### How to test

Create a MCPAR report as a state user
Verify the new field is in the DB using dynamodb admin `DYNAMO_ENDPOINT=http://localhost:8000 dynamodb-admin`
Verify the radio option hydrates in the frontend as expected

Create MLR reports and verify they are the same and create normally

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
---

### CODE REVIEW

File: services/ui-src/src/components/modals/AddEditReportModal.tsx
Comment (gmrabian) [inline services/ui-src/src/components/modals/AddEditReportModal.tsx:93]: adds field to payload for saving to backend

File: services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx
Comment (gmrabian) [inline services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx:147]: adds field to hydration data for reopening to edit

File: services/app-api/utils/types/reports.ts
Comment (karla-vm) [reply services/app-api/utils/types/reports.ts:154]: I think this works as an `AnyObject`, does this need to be an array or can it be a single object?

**Comment by BearHanded** [comment]
Do we need this behind a feature flag for the time being?

**Comment by gmrabian** [comment]
> Do we need this behind a feature flag for the time being?

That's a good question and I think the answer is yes, we do. Not sure how that would work for the backend stuff then 🤔 would we set the types to optional or would we hardcode that value to false?
That, or we could stack the next set of changes on this branch and not merge them until we're ready

**Review by benmartin-coforma — COMMENTED** [review]
I approve of the work so far, but before merging we need either a feature flag or a green light to go live.

The YoY Copy feature flag was implemented by duplicating `addEditMcparReport.json`, and choosing the old or new json depending on the feature flag. We could do something similar here, but we open ourselves up to combinatorial problems if we're not sure when YoY will go live before PCCM - that is, we would have 4 copies of the json instead of 2. (And if we don't clean these up, the next feature will have 8, and so on.)

Feature flagging on the frontend, such that it won't request PCCM behavior from the backend until go-live, allows us to merge in the backend changes unflagged, which is very nice. Not so nice is the work Garrett has put into testing the frontend - I do not know how we would get the flag into our jest & cypress tests, and I'm highly skeptical the effort would be worth it.

**Review by karla-vm — DISMISSED** [review]

**Comment by gmrabian** [comment]
Approvals for these changes were in a separate, stacked [PR](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/pull/11490). We will merge them once we reapprove and are ready for this to move into pre-prod and prod environments.

**Review by karla-vm — APPROVED** [review]

**Pull request merged**
